inc/pendingreasoncron.class.php

```diff
@@ -0,0 +1,180 @@
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * @since x.x.x
+ */
+class PendingReasonCron extends CommonDBTM
+{
+   const TASK_NAME = 'pendingreason_autobump_autosolve';
+
+   /**
+    * Get task description
+    *
+    * @return string
+    */
+   public static function getTaskDescription(): string {
+      return __("Send automated bumps on pending tickets and solve them if needed");
+   }
+
+   public static function cronInfo($name) {
+      return [
+         'description' => self::getTaskDescription(),
+      ];
+   }
+
+   /**
+    * Run from cronTask
+    *
+    * @param CronTask $task
+    */
+   public static function cronPendingreason_autobump_autosolve(CronTask $task) {
+      global $DB;
+
+      $config = Config::getConfigurationValues('core');
+
+      if (empty($config['system_user'])) {
+         Toolbox::logError("Mising system_user config");
+         return 0;
+      }
+
+      $user = User::getById($config['system_user']);
+      if (!$user) {
+         Toolbox::logError("Mising system_user user ({$config['system_user']})");
+         return 0;
+      }
+
+      $targets = [
+         new Ticket(),
+         new Change(),
+         new Problem
+      ];
+
+      $now = date("Y-m-d H:i:s");
+
+      foreach ($targets as $target) {
+         $data = $DB->request([
+            'SELECT' => 'id',
+            'FROM' => $target::getTable(),
+            'WHERE' => [
+               'status' => CommonITILObject::WAITING,
+               'auto_bump' => ['>', 0],
+            ]
+         ]);
+
+         foreach ($data as $row) {
+            /** @var CommonITILObject */
+            $item = $target::getById($row['id']);
+            if (!$item) {
+               Toolbox::logError("Failed to load {$target::getType()}::{$row['id']}");
+               continue;
+            }
+
+            $next_bump = $item->getNextAutoBumpDate();
+            $resolve = $item->getAutoResolvedate();
+
+            if ($next_bump && $now > $next_bump) {
```

Comment:
I've thought a bit about that when writing this code.

If the cron was disabled for 2 month and then re-enabled, do you want it to:
* Send all the missing bumps and resolutions at once, meaning it would bump multiple times some tickets then solve then instantly
* Only send one bump and start the next bump delay from here

I went with the second option.

Reply:
You could also send a resolution without sending missing bumps. I ticket is pending since 3 months, closing it without sending a missing bump should not be a problem.

Reply:
I don't see it that way because the auto close policy is defined by how many bump you do before closing the ticket.
If no bumps were done -> we can't solve the ticket.